src/main/webapp/script.js

```diff
@@ -236,3 +255,39 @@ function getTooltipContent(node) {
 
   return content;
 }
+
+/**
+ * When a next/previous button is clicked, modifies the mutation index of the
+ * current graph to represent the new state. Then, the corresponding
+ * graph is requested from the server.
+ */
+function navigateGraph(amount) {
+  currGraphNum += amount;
+  if(currGraphNum < 0) {
+    currGraphNum = 0;
+  }
+  if(currGraphNum >= numMutations) {
+    currGraphNum = numMutations;
+  }
+  generateGraph();
+  updateButtons();
```

Comment:
One thing to consider here is what happens when the graph takes a long time to load, and the user keeps pressing next/prev impatiently? There would be multiple calls to navigateGraph then. 

Maybe we should prevent this from happening by disabling both buttons when generate graph is happening? Maybe also have an indicator (e.g. spinning wheel)? 

Does not have to be addressed by this PR!

Reply:
I didn't know where else this could go, so I disabled both before the fetch request and enabled both before the return statement in generateGraph. Do you think there's a better way to do it, or would this be okay?

Reply:
I think this is fine! You might want to play with it locally a bit to see if this looks good to you from a user experience perspective. For example, you can manually introduce a 10 second delay to the generateGraph function and see if interaction with buttons make sense. 

Since generateGraph already calls updateButtons, delete this line?

Reply:
I found that calling generateGraph from the test file was a bit problematic because it is asynchronous but our tests run before it finishes so the buttons might just be globally disabled causing the test to fail. I changed it so that in the HTML, I call navigate graph and then generate graph (which calls update buttons) and in the tests, I call navigate graph and update buttons on button click. Does that sound okay?